src/common/transformations/src/transformations/common_optimizations/activations_scaling.cpp

```diff
@@ -0,0 +1,222 @@
+                constant_nodes.insert(node->get_friendly_name());
+                continue;
+            }
+            if ((num_const_inputs + num_normal_inputs) == node->get_input_size()) {
+                normal_nodes.insert(node->get_friendly_name());
+                continue;
+            }
+        }
+
+        if (num_scaled_down_inputs == 0) {
+            continue;
+        }
+
+        auto result = std::dynamic_pointer_cast<ov::op::v0::Result>(node);
+        if (result && num_scaled_down_inputs == 1) {
+            auto dep = node->input(0);
+            std::shared_ptr<ov::Node> scale_const = (dep.get_element_type() == ov::element::f16) ?
+                                                    scale_const_f16 : scale_const_f32;
+            auto scale_up = std::make_shared<ov::op::v1::Multiply>(dep.get_source_output(), scale_const->output(0));
+            dep.replace_source_output(scale_up->output(0));
+        }
+
+        //    input0         input1            input0         input1
+        // (scaled_down)  (non-scaled)      (scaled_down)  (non-scaled)
+        //          \         /                      \         /
+        //           \       /         ==>            \    scale_down
+        //            \     /                          \     /
+        //              add                              add
+        auto add = std::dynamic_pointer_cast<ov::op::v1::Add>(node);
+        if (add && num_scaled_down_inputs == 1) {
+            for (auto& dep: node->inputs()) {
+                if (scaled_down_nodes.find(dep.get_source_output().get_node_shared_ptr()->get_friendly_name()) == scaled_down_nodes.end()) {
+                    std::shared_ptr<ov::Node> inverse_scale_const = (dep.get_element_type() == ov::element::f16) ?
+                                                                    inverse_scale_const_f16 : inverse_scale_const_f32;
+                    auto scale_down = std::make_shared<ov::op::v1::Multiply>(dep.get_source_output(),
+                                                                             inverse_scale_const->output(0));
+                    dep.replace_source_output(scale_down->output(0));
+                }
+            }
+        }
+
+        auto multiply = std::dynamic_pointer_cast<ov::op::v1::Multiply>(node);
+        auto matmul = std::dynamic_pointer_cast<ov::op::v0::MatMul>(node);
+        if ((multiply || matmul) && num_scaled_down_inputs == 2) {
+            auto dep = node->input(1);
+            std::shared_ptr<ov::Node> scale_const = (dep.get_element_type() == ov::element::f16) ?
+                                                    scale_const_f16 : scale_const_f32;
+            auto scale_up = std::make_shared<ov::op::v1::Multiply>(dep.get_source_output(), scale_const->output(0));
+            dep.replace_source_output(scale_up->output(0));
+        }
+
+        auto sdpa = std::dynamic_pointer_cast<ov::op::v13::ScaledDotProductAttention>(node);
```

Comment:
[random spot] Please add tests for the transformation

Reply:
I added unit tests.